clustered/server/src/test/java/org/ehcache/clustered/server/store/impl/ReferenceStoreImpl.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.clustered.server.store.impl;
+
+import org.ehcache.clustered.common.store.Chain;
+import org.ehcache.clustered.common.store.Element;
+import org.ehcache.clustered.server.store.ServerStore;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+/**
+ * Implements {@link ServerStore}
+ */
+public class ReferenceStoreImpl implements ServerStore  {
+
+  private final Map<Long, Chain> map = new HashMap<Long, Chain>();
+  private final List<ReadWriteLock> locks = new ArrayList<ReadWriteLock>();
+  private final AtomicLong sequenceGenerator = new AtomicLong();
+
+  private final int LOCK_COUNT = 16;
+
+  public ReferenceStoreImpl() {
+    for (int i = 0; i < LOCK_COUNT; i++) {
+      locks.add(new ReentrantReadWriteLock());
+    }
+  }
+
+  private ReadWriteLock getLock(long key) {
+    return locks.get((int)key%LOCK_COUNT);
+  }
+
+  @Override
+  public Chain get(long key) {
+    Lock lock = getLock(key).readLock();
+    try {
+      lock.lock();
```

Comment:
It's harmless here since you're only using `j.u.c.l.ReentrantReadWriteLock` but your pattern is wrong: the lock should be taken out of the `try - catch` block.